app/core/api/thang-types.js

```diff
@@ -0,0 +1,35 @@
+const fetchJson = require('./fetch-json')
+const _ = require('lodash')
+
+/**
+ * @typedef getThangOptions
+ * @type {object}
+ * @property {string} slug - The slug of the thangtype
+ * @property {string[]} [projection] - Projections that can be applied.
+ */
+
+/**
+ * Retrieves a thangType from the database.
+ * @param {getThangOptions} options - Support projection field with a list of string attributes.
+ */
+export const getThang = async options => {
+  const data = {}
+  if (options.project) {
```

Comment:
`options && options.project` ?

Reply:
What if `options.slug` is not defined? Is that handled in the server API?

Reply:
Great catch. Adding validation.